BitList.cs

```diff
@@ -0,0 +1,1553 @@
+    /// Copy this instance to the specified array.
+    /// </summary>
+    /// <param name="arrayIndex">Start position.</param>
+    public void CopyTo(ushort[] array, int arrayIndex)
+    {
+        ToUInt16Array().CopyTo(array, arrayIndex);
+    }
+
+    /// <summary>
+    /// Copy this instance to the specified array.
+    /// </summary>
+    /// <param name="arrayIndex">Start position.</param>
+    public void CopyTo(uint[] array, int arrayIndex)
+    {
+        ToUInt32Array().CopyTo(array, arrayIndex);
+    }
+
+    /// <summary>
+    /// Copy this instance to the specified array.
+    /// </summary>
+    /// <param name="arrayIndex">Start position.</param>
+    public void CopyTo(ulong[] array, int arrayIndex)
+    {
+        ToUInt64Array().CopyTo(array, arrayIndex);
+    }
+
+    /// <summary>
+    /// Copy this instance to the specified array.
+    /// </summary>
+    /// <param name="arrayIndex">Start position.</param>
+    public void CopyTo(float[] array, int arrayIndex)
+    {
+        ToSingleArray().CopyTo(array, arrayIndex);
+    }
+
+    /// <summary>
+    /// Copy this instance to the specified array.
+    /// </summary>
+    /// <param name="arrayIndex">Start position.</param>
+    public void CopyTo(double[] array, int arrayIndex)
+    {
+        ToDoubleArray().CopyTo(array, arrayIndex);
+    }
+
+    /// <summary>
+    /// Copy this instance to the specified array.
+    /// </summary>
+    /// <param name="arrayIndex">Start position.</param>
+    public void CopyTo(decimal[] array, int arrayIndex)
+    {
+        ToDecimalArray().CopyTo(array, arrayIndex);
+    }
+
+    /// <summary>
+    /// Copy this instance to the specified array.
+    /// </summary>
+    /// <param name="arrayIndex">Start position.</param>
+    public void CopyTo(char[] array, int arrayIndex)
+    {
+        ToCharArray().CopyTo(array, arrayIndex);
+    }
+
+    public bool[] ToBooleanArray()
+    {
+        if (IsEmpty)
+            return new bool[0];
+        else
+        {
+            var bools = new bool[bits.Length];
+            for (int i = 0; i < bits.Length; i++)
+                bools[i] = bits[i];
+            return bools;
+        }
+    }
+
+    public byte[] ToByteArray()
+    {
+        if (IsEmpty)
+            return new byte[0];
+        else
+        {
+            int inum = bits.Length / 8;
+            double dnum = (double)bits.Length / 8;
+            if (dnum.ToString().Contains(".") || dnum.ToString().Contains(","))
+                inum++;
+            var bytes = new byte[inum];
+            bits.CopyTo(bytes, 0);
+            return bytes;
+        }
+    }
+
+    public short[] ToInt16Array()
+    {
+        if (IsEmpty)
+            return new short[0];
+        else
+        {
+            var bytes = ToByteArray();
+            var bytesList = new List<byte[]>();
+            var c = -1;
+            for (int i = 0; i < bytes.Length; i++)
+            {
+                c++;
+                if (i == 0 | c == 2)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Correct this '|' to '||'.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5545753)